Cleanup of README

arves100 opened the pull request:
- Added Windows build instructions (vcpkg+choco)
- Removed old library requirements

UndarkAido commented on README.md:
```diff
@@ -9,19 +9,23 @@
 ### Install Prerequisites
 #### Install Git
- * If you want to use Git in the command line, run `sudo apt install git`
+ * If you want to use Git in the command line, run `sudo apt install git`, you can get [Git for Windows](https://git-scm.com/) or use `choco install git`
```
Clarify platforms

UndarkAido replied:
Something like "run `sudo apt install git` on Ubuntu or on Windows download git from XX or run XX in the command line if you have chocolatey"

UndarkAido commented on README.md:
```diff
@@ -9,19 +9,23 @@
     * `clang` is an alternative that I feel has better error statements, you can install it with `sudo apt install clang`
- 2. Install CMake with `sudo apt install cmake`
+    * The typical standard compiler for Windows is [Microsoft Visual C++](https://www.visualstudio.com/)
+    * [MinGW](https://www.msys2.org/) or [Cygwin](https://www.cygwin.com/) are valid alternatives if you like GCC, oh there's also [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10) as well ^^.
```
It isn't clear this line applies to Windows

UndarkAido commented on README.md:
```diff
@@ -9,19 +9,23 @@
+    * The typical standard compiler for Windows is [Microsoft Visual C++](https://www.visualstudio.com/)
+    * [MinGW](https://www.msys2.org/) or [Cygwin](https://www.cygwin.com/) are valid alternatives if you like GCC, oh there's also [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10) as well ^^.
+	
+ 2. Install CMake with `sudo apt install cmake` or by downloading it from [CMake](https://cmake.org/download/) or get it with `choco install cmake`.
```
Clarify platforms

UndarkAido commented on README.md:
```diff
@@ -9,19 +9,23 @@
- * Fedora/RHEL-based `sudo dnf install openssl-devel boost-devel cryptopp-devel cryptopp`
+ * Debian-based (Ubuntu) `sudo apt install build-essential libboost-all-dev libssl-dev`
+ * Fedora/RHEL-based `sudo dnf install openssl-devel boost-devel`
+ * Windows Get [vcpkg](https://github.com/microsoft/vcpkg), follow the page instruction to install it, then type`vcpkg install boost-system boost-beast openssl`
```
Grammar here feels awkward. Maybe we need a `tools` section for things like `git`, `vcpkg`, `cmake`, and `choco`?

UndarkAido replied:
Oh, and maybe something about how `vcpkg` is cross-platform?

arves100 replied:
Should I just point vcpkg in all platforms or say something like you can use vcpkg on linux/mac as well?

UndarkAido replied:
That would be good

UndarkAido commented on README.md:
```diff
@@ -30,7 +34,9 @@
 #### Clone the Code
-Run `git clone --recursive https://github.com/discordpp/echo-bot.git` in your desired directory. If you made a fork, replace `discordpp` and `echo-bot` as applicable.
+Run `git clone --recursive https://github.com/discordpp/echo-bot.git` in your desired directory.
+Make sure to run `git submodule update --init --recursive` in the new echo-bot directory to make sure it downloads all submodules.
```
`git submodule update --init --recursive` is not required if you `clone` with `--recursive`

UndarkAido commented on README.md:
```diff
@@ -55,8 +61,8 @@ Run `git clone --recursive https://github.com/discordpp/echo-bot.git` in your de
- 4. Build the makefiles with `cmake ../`
- 5. Build the bot with `make`
+ 4. Build the makefiles with `cmake ../` (If you are using vcpkg then `cmake -DCMAKE_TOOLCHAIN_FILE="your vcpkg.cmake folder" ../`)
+ 5. Build the bot with `make` (if you are using Visual Studio just open `echo_bot.sln` and build from there)
```
Remove "just"

UndarkAido commented on README.md:
```diff
@@ -55,8 +61,8 @@ Run `git clone --recursive https://github.com/discordpp/echo-bot.git` in your de
  3. Go back to your Application's 'bot' page from before, and copy the token. Run `echo Bot [token] > token.dat`, replacing the brackets and the word token
- 4. Build the makefiles with `cmake ../`
- 5. Build the bot with `make`
+ 4. Build the makefiles with `cmake ../` (If you are using vcpkg then `cmake -DCMAKE_TOOLCHAIN_FILE="your vcpkg.cmake folder" ../`)
```
Maybe add the `vcpkg` stuff as an indented subpoint of this one instead of in parenthesis?

arves100 commented:
Ok, I've pushed a new commit with the new README.

- Moved vcpkg cmake command into a subpoint
- Removed git submodule update
- Moved cmake, vcpkg istructions into the Prerequisites
- I guess I explained what vcpkg is and that is required for Windows, and how to get it
- Updated the compiler to clarify the platforms

Let me know if there's anything else I need to modify.

UndarkAido commented on README.md:
```diff
@@ -9,19 +9,28 @@
+ * Windows Download [CMake](https://cmake.org/download/) or run `choco install cmake` if you have Chocolatey installed.
+#### Install vcpkg
+ * Vcpkg is a cross platform library manager, you can install it by reading the instructions on this [page](https://github.com/microsoft/vcpkg)
+ * This program is required to build echo-bot under Windows.
```
Refer to Echo with the proper capitalized name at the top of the repo, not the constriined-formatting repo name

UndarkAido commented on README.md:
```diff
@@ -9,19 +9,28 @@
+    * the typical standard on Linux is `gcc`, you can install it on Ubuntu with `sudo apt install gcc`
+    * `clang` is an alternative that I feel has better error statements, you can install it on Ubuntu with `sudo apt install clang`
+    * The typical compiler on Windows is [Microsoft Visual C++](https://www.visualstudio.com/)
+    * [MinGW](https://www.msys2.org/) or [Cygwin](https://www.cygwin.com/) are valid alternatives if you like GCC running on Windows, oh there's also [WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10) as well ^^.
```
Maybe clarify a tiny bit what WSL does? Something like "WSL, which allows you to use a Linux development environment on Windows"

UndarkAido replied:
Replace 'oh' with 'and'?

UndarkAido commented on README.md:
```diff
@@ -9,19 +9,28 @@
+ * If you want to use Git in the command line, run `sudo apt install git` on Ubuntu, on Windows you can download [Git for Windows](https://git-scm.com/) or use `choco install git` if you have Chocolatey installed.
  * If you're less terminal-inclined, you can install a Git GUI wrapper. [There's a nice list here.](https://git-scm.com/downloads/guis/)
     * I would personally recommend [GitKraken](https://www.gitkraken.com/). The free version has everyhting you need, (although their conflict resolver for Pro is nice,) and you can also get Pro free with [GitHub Education](https://education.github.com/)!
+#### Install CMake
```
Include <sup>AND</sup>

UndarkAido commented on README.md:
```diff
@@ -9,19 +9,28 @@
+ * Debian-based (Ubuntu) run `sudo apt install cmake`
+ * Fedora/RHEL-based `sudo dnf install cmake`
+ * Windows Download [CMake](https://cmake.org/download/) or run `choco install cmake` if you have Chocolatey installed.
+#### Install vcpkg
```
Include something like <sup>OPTIONALLY</sup>

UndarkAido commented on README.md:
```diff
@@ -30,7 +39,8 @@
 #### Clone the Code
-Run `git clone --recursive https://github.com/discordpp/echo-bot.git` in your desired directory. If you made a fork, replace `discordpp` and `echo-bot` as applicable.
+Run `git clone --recursive https://github.com/discordpp/echo-bot.git` in your desired directory.
+If you made a fork, replace `discordpp` and `echo-bot` as applicable.
```
Why replace `discordpp`?

UndarkAido commented on README.md:
```diff
@@ -9,19 +9,28 @@
+ * Vcpkg `vcpkg install boost-system boost-beast openssl`
 #### <sup>OR</sup> Install Everything at Once
-`sudo apt install git clang cmake build-essential libboost-all-dev libssl-dev libcrypto++-dev libcurl4-openssl-dev`
+ * Debian-based `sudo apt install git clang cmake build-essential libboost-all-dev libssl-dev`
```
Add (Ubuntu) like above

UndarkAido commented on README.md:
```diff
@@ -8,29 +8,29 @@
 
 ### Install Prerequisites
-#### Install Git
+#### <sup>AND</sup>Install Git
```
It doesn't make sense for the first point to have an 'AND'

arves100 commented:
Updated.
